regression/Makefile

```diff
@@ -27,6 +27,10 @@ DIRS = cbmc \
        contracts \
        # Empty last line
 
+ifeq ($(WITH_MEMORY_ANALYZER),1)
+  DIRS += memory-analyzer
+endif
```

Comment:
Do we really need to make this conditional? Can't we just make this dependent on the OS/`BUILD_ENV_`? At least I'd much prefer to keep things as simple as possible.

Reply:
@tautschnig We could enable the flag by default on Linux if you'd prefer, but the memory analyzer (and its tests!) don't actually work on some platforms (including some linux configurations), so I think just having a single flag saying that you want it is already the "simple" version.

Reply:
What is it that keeps is from working across all Linux configurations? I'm worried that anything that is optional will die at some point, and even when it doesn't people might not give it a whole lot of testing in their local builds. So, if we can, we should just enable it by default.

Reply:
It's slightly exotic, but some Linux systems have `ptrace` disabled or heavily restricted: https://www.kernel.org/doc/Documentation/security/Yama.txt - I believe we had at least one of the CI environments here configured this way, so `memory-analyzer` actually didn't work in CI for a while. Slightly less exotic is the possibility of people not having `gdb` installed.

Again, I don't mind turning it on by default on Linux as it's going to work _most_ of the time, but I'd also like to keep a user controlled option to turn it off if they don't want it, or forcing it to turn on even if our platform detection logic thinks it wouldn't work.